sci-physics/geant-vmc/geant-vmc-4.5.1_p1.ebuild

```diff
@@ -0,0 +1,113 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake
+
+if [[ ${PV} == *9999* ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/vmc-project/geant4_vmc.git"
+else
+	MY_PV=$(ver_rs 1-2 - $(ver_cut 2-))
+	SRC_URI="https://github.com/vmc-project/geant4-vmc/archive/v${MY_PV}.tar.gz -> ${P}.tar.gz"
+	S="${WORKDIR}/geant4_vmc-${MY_PV}"
+	KEYWORDS="~amd64 ~x86"
+fi
+
+DESCRIPTION="Virtual Monte Carlo Geant4 implementation"
+HOMEPAGE="http://root.cern.ch/root/vmc/VirtualMC.html"
+
+LICENSE="GPL-3"
+SLOT="4"
+IUSE="+c++11 c++14 c++17 doc examples geant3 +g4root +mtroot rootvmc vgm test"
+
+REQUIRED_USE="^^ ( c++11 c++14 c++17 )"
+
+RDEPEND="
+	>=sci-physics/geant-4.10.06:=[opengl,geant3?]
+	rootvmc? (
+		>=sci-physics/root-6.18:=[vmc]
+		!!sci-physics/vmc
+	)
+	!rootvmc? (
+		>=sci-physics/root-6.18:=[-vmc]
+		sci-physics/vmc:=
+	)
+	c++11? (
+		>=sci-physics/geant-4.10.06[c++11]
+		>=sci-physics/root-6.14:=
+		sci-physics/vmc[c++11]
+		vgm? ( >=sci-physics/vgm-4.8:=[c++11] )
+	)
+	c++14? (
+		>=sci-physics/geant-4.10.06[c++14]
+		>=sci-physics/root-6.14:=
+		sci-physics/vmc[c++14]
+		vgm? ( >=sci-physics/vgm-4.8:=[c++14] )
+	)
+	c++17? (
+		>=sci-physics/geant-4.10.06[c++17]
+		>=sci-physics/root-6.14:=
+		sci-physics/vmc[c++17]
+		vgm? ( >=sci-physics/vgm-4.8:=[c++17] )
+	)"
+DEPEND="${RDEPEND}
+	doc? ( app-doc/doxygen )"
+RESTRICT="
+	!examples? ( test )
+	!geant3? ( test )
+	!g4root? ( test )
+	!mtroot? ( test )
+	!test? ( test )
+	!vgm? ( test )"
+
+DOCS=(history README.md)
+
+src_configure() {
+	local mycmakeargs=(
+		-DGeant4VMC_USE_VGM="$(usex vgm)"
+		-DGeant4VMC_USE_GEANT4_G3TOG4="$(usex geant3)"
+		-DGeant4VMC_USE_G4Root="$(usex g4root)"
+		-DGeant4VMC_BUILD_MTRoot="$(usex mtroot)"
+		-DGeant4VMC_BUILD_EXAMPLES="$(usex test)"
+		-DGeant4VMC_INSTALL_EXAMPLES="$(usex examples)"
+	)
+	cmake_src_configure
+}
+
+src_compile() {
+	cmake_src_compile
+	if use doc ; then
+		local dirs=(
+			source
+			$(usev g4root)
+			$(usev mtroot)
+			$(usev examples)
+		)
+		local d
+		for d in "${dirs[@]}"; do
+			pushd "${d}" > /dev/null || die
+			doxygen || die
+			popd > /dev/null || die
+		done
+	fi
+}
+
+src_test() {
+	# Required by sci-physics/root for pointer validity checking,
+	# see e.g. https://sft.its.cern.ch/jira/browse/ROOT-8146 .
+	addwrite /dev/random
```

Comment:
Ha, good reminder for me! In recent versions of ROOT this is no longer necessary! See root-project/root@8f8c786cc491438548da4a29385cca4f470d2d29 and root-project/root@35bb351083b35ce64aae43cb29e09f3b8bce0262 :-)

Reply:
Wow, that's really nice!

I stumbled upon this quite a while ago, when `sci-physics/root` was less well-maintained:
https://sft.its.cern.ch/jira/browse/ROOT-8146
While in that issue I assumed the performance implications were not too heavy, a Bonn colleague of mine apparently found otherwise in the meantime (I did not see it, since we compile almost all code), I completely missed that:
https://sft.its.cern.ch/jira/browse/ROOT-9936
So I'm really happy about the new solution. Thanks for implementing it! :+1:

Reply:
The pointer check mechanism is still hitting performance, just not as severely as before.

Reply:
Thanks! While this is now getting a bit off-topic, I remember that at some point I asked (Axel, I think?) if it might be a good idea to have a way to use {{TCling::Declare}} and explicitly turn off the check for the jitted code (for performance optimization when you know what your code is doing). It seems that's not in yet, if you want to follow up, we could take this to an upstream issue?